src/hall.scad

```diff
@@ -0,0 +1,60 @@
+
+module hall (hallSize)
+
+{
+    hallSize = hallSize;        //Dimension of the side of the square hall.
+    cf = 0.75;                  //chamfer factor for the corner of the square hall.
+    hf = 1;                     //Ratio of the height of the wall to the hallSize.
+    hhf = 1.25*hf;              // Ratio of the height of the 4 rectangular block to the height of the square hall. 
+    lbf = 0.80;                 //Ratio of the length of the 4 rectangular block to the hallSize.
+    tf = 0.10;                  //Ratio of the thickness the 4 rectangular block to the hallSize.
+    dg = 0.05;                  //Ratio of the depth of gutter and hallSize found on the top of the hall. 
```

Comment:
Some of these values can be made as module parameters so that we have minimal hardcoded values. However, you can keep the default values for the parameters so that when using the module, we don't have to specify all the parameters but still gives us the ability to override all of them if needed.

To declare parameters with default values:
```
module module_name(param_1=0.75, param_2=1.5, param_3=1) {
    ...
}
```